### ISSUE TITLE
A couple of bugs with profiles

### ISSUE THREAD

**Issue by Uukrull** [asciidisco/plugin.video.netflix#16]
I'm using the latest addon version: https://github.com/asciidisco/plugin.video.netflix/commit/dfc7738380f5ce833063ce74e2c0ab9b82b84eae, a fresh build compiled for Windows from the latest kodi-agile: https://github.com/FernetMenta/kodi-agile/commit/040db824a3760ee941ef625129dc26c9d6142f17 and the latest inpustream: https://github.com/liberty-developer/inputstream.adaptive/commit/764de78a262091b2bd050b97778d0f30e6501460

Issue 1:
My Netflix account has two profiles. Profile 1 has the language set to Spanish and Profile 2 to English. If I use Profile 2, the categories menu (My List, Continue Watching,...) is in English:
![image](http://i.imgur.com/6lwJM3I.jpg)
but the titles and descriptions are in Spanish:
![image](http://i.imgur.com/KGZTKle.jpg)

If I change the Profile 1 language to English, I can see the titles and descriptions in English with Profile 2 so, clearly, the Profile 2 always uses whatever language Profile 1 has.

Issue 2:
If I select Profile 2 from the profiles selection screen, go back to this screen and select it again, the profile screen I get is the one from Profile 1.

These issues can be easily reproduced but I can provide a log if needed.

Thanks in advance.

**Comment by asciidisco** [comment]
@Uukrull Thank you for the feedback (and sorry for the delayed in reply).

Regarding Issue No. 1:
We´re not parsing the language per user profile atm., the language is extracted from the account owner profile. I must admit, I wasn't aware that it is possible to have different languages using one account. Fixing that is now on our list.

Regarding Issue No. 2:
Great observation. That´s a bug in the in memory caching layer, the caching id isn't generated on a per user base, but only on a per category base. Refactoring the cache is the next thing on my list, after we fixed the big problem with the crashing background service that still keeps us occupied.


Btw.: If you do speak spanish, would you mind contributing a spanish gettext file to the project. There are only around 30 messages we need to translate & that would definitely help us :)

Cheers

**Comment by Uukrull** [comment]
> Btw.: If you do speak spanish, would you mind contributing a spanish gettext file to the project. There are only around 30 messages we need to translate & that would definitely help us :)

Done: https://github.com/Uukrull/plugin.video.netflix/commit/422bc2d2d5516307c6a451c3508c24d991412192 Do I open a PR?

Many thanks to you and all the people involved for bringing Netflix to Kodi. I never thought I could see this day :)

**Comment by asciidisco** [comment]
@Uukrull Awesome. Yes, please PR it :)

**Comment by asciidisco** [comment]
@Uukrull I pushed a release a couple of moments ago. Issue No. 2 should be resolved.
I also looked briefly into the causes of the behaviour of Issue No. 1, but I can't seem to identify how Netflix does determine the language of the user. I have the feeling that they´ll do it with some cookie magic when calling the "/browse" route (which we don't when switching profiles, as this is really slow & there´s no fresh information for us to use). We only called once at the first startup. I guess that would explain your observations.

I´ll do some more research & curl magic to see if I can reproduce this on the lowest level possible, maybe the trick is for us the fake some cookie setting or such.

**Comment by Uukrull** [comment]
> Issue No. 2 should be resolved.

Yes, it is. Many thanks!

**Comment by asciidisco** [comment]
And yes, it is cookie magic, done by a cookie set from the browse HTML page (which takes ages to load & does only accept GET requests, so no chance to do a HEAD here & fetch it faster).

The cookie looks like this:
```
lhpuuidh-browse-ZFFNPO5JSVGUHB4WTKI6UYYBSD=DE%3AEN%3A4b59b7af-29a8-482d-a6b4-137c3303ebb8_ROOT;
```

a) the `ZFFNPO5JSVGUHB4WTKI6UYYBSM` part of the cookie name is your profiles GUID.
b) the `urlencoded(DE:EN:)` part of the cookie contents defines the accounts origin & the selected language (origin DE & language EN here in my case), the second part is the so called "lolomo" id (`b59b7af-29a8-482d-a6b4-137c3303ebb8_ROOT`), which Netflix assigns to your root account view (the one is needed to get your accounts "Suggestions", "Genres", "Watchlist" etc.)

So, long story short: 
Fetching that cookie takes too long, we shouldn't go down that route.
Faking it would be more interesting (we have all the data, EXCEPT the language), so basically, we could get the users language settings from Kodi & always fetch the language based on the current Kodi settings.
In the End, if you share your Kodi installation with other people at home, you most probably end up with different profiles that have their own settings in Kodi (including the interface language) as well.

**Comment by Uukrull** [comment]
Great! That could also explains why I get Latin American Spanish (es) audio tracks instead of the Castilian (es-ES) ones (my account was created in the USA before Netflix arrived to Spain) with the addon.

**Comment by trummerjo** [comment]
i think that actually the audio stream with the language selected in kodi should be selected from the stream? Is there a diffrent setting in kodi for es and es-ES?

**Comment by Uukrull** [comment]
This isn't a Kodi problem because I tried a Netflix account created in Spain and it shows the correct Castilian audio track. For example, all the Netflix Originals have the Spanish audio track in Latin American Spanish instead of Castilian if I use the addon, but I have no problems with a browser or the Shield TV app (that shows in the information screen Country = ES and Language = en-US).

Could it be possible to add a hidden setting to generate that fake cookie with the country and language we want?

**Comment by hawkeyexp** [comment]
> Issue 2:
If I select Profile 2 from the profiles selection screen, go back to this screen and select it again, the profile screen I get is the one from Profile 1.

Problem is still present.
